### ISSUE TITLE
Parsing error: ',' expected

### ISSUE THREAD

**Issue by memao** [typescript-eslint/typescript-eslint#1716]
**What code were you trying to parse?**
```ts
// Assert
        expect(inspectorResponse.props).toEqual({
            children: [
                { horizontalAlign: "center", children: [{ children: mockChild }] },
                { children: [{ children: React.createElement(Heading, { "tag": HeadingTag.h2, "size": HeadingSize._5, "children": "Stories" }) }] },
                { children: [{ children: React.createElement(Typography, { "tag": TypographyTag.h2, "size": TypographySize._3, "children": "Test" }) }] },
                { children: [{ children: React.createElement(ActionTrigger, { "children": "Test" }) 
            ]
        });
```

**What did you expect to happen?**
Don't show any error

**What actually happened?**
error  Parsing error: ',' expected at the last 3 lines of the code.

**Versions**

| package                     | version |
| --------------------------- | ------- |
| `@typescript-eslint/parser` | `2.22.0` |
| `TypeScript`                | 3.6.3` |
| `ESLint`                    | `6.8.0` |
| `node`                      | `10.16.3` |
| `npm`                       | `6.13.2` |

**Comment by bradzacher** [comment]
This is the correct error.
Your code is invalid.
This is why you should avoid long complex lines like that.
Check the bracket matching on your last line.

**Comment by memao** [comment]
yeah, agreed.  This code is invalid. I guess my question is that this error is still showing up even when there is no rule in the ESLint config. So I am wondering which rule is looking for this type of error? Just to under how typescript-eslint is working.

**Comment by bradzacher** [comment]
ESLint parses the files you tell it to run on.
If you're using an IDE extension, this is also true - the extension essentially just does a cli run under the hood.

Just because there are no rules, doesn't mean there's nothing for ESLint to check.
For example, you might have a [configuration comment](https://eslint.org/docs/user-guide/configuring#using-configuration-comments) in one of the files that turns on a rule, which ESLint has to respect, even if there's no rules configured in the eslintrc.

**Comment by memao** [comment]
@bradzacher We don't have any inline comment to enable ESLint rules. We also have "noInlineConfig" set to "true" in .eslintrc.json. 
I looked through typescript-eslint/typescript-estree, found it choose to show a white list of errors from typescript to users as in semantic-or-syntactic-errors.ts. However  I couldn't find the specific error I see "Parsing error: ',' expected" in the white list. 
I am wondering if you could please help me understand where this error come from

**Comment by bradzacher** [comment]
This error comes from typescript, as we use that to parse the file.

Note that our tooling just parses the files eslint tells us to.

You'll have to talk to the eslint folks, or dig into the eslint codebase if you want to find out about how the parsing is coordinating.

My guess is that it still parses the file despite the fact that you have the flag set.
